backend/controller/dal/database_resolver.go

```diff
@@ -0,0 +1,81 @@
+package dal
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/TBD54566975/ftl/backend/controller/sql"
+	"github.com/TBD54566975/ftl/common/configuration"
+	"github.com/alecthomas/types/optional"
+)
+
+// DatabaseResolver loads values a project's configuration from the given database.
+type DatabaseResolver struct {
+	db sql.DBI
+}
+
+// DatabaseResolver should only be used for config, not secrets
+var _ configuration.Resolver[configuration.Configuration] = DatabaseResolver{}
+
+func (d DatabaseResolver) Role() configuration.Configuration { return configuration.Configuration{} }
+
+func (d *DAL) NewConfigResolver() DatabaseResolver {
+	return DatabaseResolver{
+		db: d.db,
+	}
+}
+
+func (d DatabaseResolver) Get(ctx context.Context, ref configuration.Ref) (*url.URL, error) {
+	module, moduleOk := ref.Module.Get()
+	if !moduleOk {
+		module = "" // global
+	}
+	accessor, err := d.db.GetConfig(ctx, module, ref.Name)
+	if err != nil {
+		// If we could not find this config within the module, then try getting
+		// again with scope set to global
+		if moduleOk {
```

Comment:
I think we should push this down to the query so we can retrieve a global value or the module specific value in a single query, something like this:

```sql
SELECT value FROM module_config
WHERE
  (module IS NULL AND name = @name)
  OR (module = @module AND name = @name)
ORDER BY module NULLS LAST -- TIL about NULLS LAST, handy
LIMIT 1
```

Reply:
Though as mentioned elsewhere we should not include the value in the URL, in which case the resolver probably doesn't even need to hit the database at all it can just return `db://[<module>.]<name>` - but this select would be used by the provider.

Reply:
It looks like there aren't any functions that take a `key *url.URL` and don't also get a `ref`, which already has the module and config name as structured data. In that case, can we just have the URL be effectively a meaningless stub and always operate off the ref?

Reply:
The biggest challenge seems to be the fact that `DatabaseResolver.Set` takes a URL as its only way to provide a value to `SetConfig` with. So we need to be able to retrieve the underlying value from the URL. I'll go with base64 encoding it for now...

Reply:
If we go with `db://[<module>.]<name>`, then what should `DatabaseResolver.Set` do? It only receives the ref and URL; no []byte value.

Reply:
Done! Resolving this thread to consolidate discussion on the other one